melodies_monet/util/region_select.py

```diff
@@ -0,0 +1,320 @@
+
+    Parameters
+    ----------
+    data : xr.Dataset
+        data to be masked
+    domain_type : str
+        type of data. Used to decide which function to apply. Should begin with
+        "custom".
+    domain_info : dict
+        Dictionary containing relevant information on domain, like url, name, etc.
+    **kwargs:
+        Extra kwargs to pass to regionmask
+
+    Returns
+    -------
+    xr.Dataset
+        masked Dataset
+    """
+    if "custom" not in domain_type:
+        raise ValueError("If regionmask is used, the domain_type should be starting with 'custom'")
+    if "polygon" in domain_type:
+        masked_data = create_custom_mask(data, domain_info["mask_info"])
+    elif "defined-region" in domain_type:
+        name_regiontype = domain_info["name_regiontype"]
+        region = domain_info["region"]
+        masked_data = create_predefined_mask(data, name_regiontype, region)
+    elif "file" in domain_type:
+        params = domain_info
+        params["mask_path"] = domain_info.get("mask_path", None)
+        params["mask_url"] = domain_info.get("mask_url", None)
+        params["region_name"] = domain_info.get("region_name", None)
+        params["abbrevs"] = domain_info.get("abbrevs", "_from_name")
+        masked_data = create_shapefile_mask(data, **params, **kwargs)
+    else:
+        raise ValueError(
+            "Could not identify the type of domain. Should be 'polygon',"
+            + " 'defined-region' or 'file'"
+        )
+    return masked_data
+
+
+def create_autoregion(data, domain_type, domain_name, domain_info=None):
+    """Selects a region using predefined boundaries.
+
+    Parameters
+    ----------
+    data : xr.Dataset | pd.DataFrame
+        data to be masked
+    domain_type : str
+        type of data. Used to decide which function to apply.
+        If domain_type == 'auto-region:', domain_info is required.
+    domain_name : str
+        This is used as the region name, or to read the info.
+    domain_info: None | dict[str, tuple[float, float, float, float]]
+        if not None, dict containing the domain name and a tuple with
+        latmin, lonmin, latmax, lonmax. Only required if domain_type
+        is auto-region:
+    Returns
+    -------
+    xr.Dataset | pd.DataFrame
+        Data as it was provided to the function
+    """
+    auto_region_id = domain_type.split(":")[1].lower()
+    if auto_region_id == "epa":
+        bounds = get_epa_region_bounds(acronym=domain_name)
+    elif auto_region_id == "giorgi":
+        bounds = get_giorgi_region_bounds(acronym=domain_name)
+    elif auto_region_id == "box":
+        bounds = domain_info["bounds"]
+    else:
+        raise ValueError(
+            "Currently, auto-region selections whithout a domain query have only "
+            "been implemented for Giorgi and EPA regions. You asked for "
+            f"{domain_type!r}. If you need more capabilities, checkout the custom:"
```

Comment:
```suggestion
            f"{domain_type!r}. If you need more capabilities, check out the custom: "
```
I would think there should be a space after "custom:", but not sure.

Reply:
Yes, I think you are right